suite/tests/pthreads/ptsig.c

```diff
@@ -45,35 +45,44 @@
 #include <unistd.h>
 #include <assert.h>
 #include <setjmp.h>
+#include "thread.h"
 
-volatile double pi = 0.0;  /* Approximation to pi (shared) */
-pthread_mutex_t pi_lock;   /* Lock for above */
-volatile double intervals; /* How many intervals? */
+volatile double pi = 0.0;         /* Approximation to pi (shared) */
+pthread_mutex_t pi_lock;          /* Lock for above */
+pthread_mutex_t sig_counter_lock; /* Lock to increment signal counter */
+volatile double intervals;        /* How many intervals? */
 static SIGJMP_BUF mark;
+static int signal_received = 0;
 
 static void
 signal_handler(int sig, siginfo_t *siginfo, ucontext_t *ucxt)
 {
+    sigcontext_t *sc = SIGCXT_FROM_UCXT(ucxt);
+    void *pc = (void *)sc->SC_XIP;
```

Comment:
Not sure why these 2 lines were moved?

Reply:
Was a bug that the "case SIGSEGV" needs pc if VERBOSE is on. I moved the 2 lines into the #if VERBOSE ifdef.